app/src/main/java/br/com/catbag/gifreduxsample/ui/components/FeedComponent.java

```diff
@@ -58,29 +57,33 @@ public FeedComponent(Context context, AttributeSet attrs, int defStyleAttr) {
 
     @Override
     public void view() {
-        final List<Gif> gifs = mGifs;
         Recycler.view(() -> {
             if (mLayoutManager == null) {
                 mLayoutManager = new LinearLayoutManager(getContext(),
                         LinearLayoutManager.VERTICAL, false);
             }
             Recycler.layoutManager(mLayoutManager);
             Recycler.hasFixedSize(true);
-            Recycler.adapter(Recycler.Adapter.simple(gifs, (viewHolder) -> {
-                renderGifView(gifs.get(viewHolder.getAdapterPosition()));
-            }));
+            if (mGifsAdapter == null) {
+                mGifsAdapter = gifsAdapter(mGifs, (gif) -> renderGifView(gif));
+                Recycler.adapter(mGifsAdapter);
+            }
+            else if (!mGifs.equals(mGifsAdapter.getGifs())) {
```

Comment:
You correct and the pattern is this you saying. I don't know why code inspect don't catch this.